src/realm/query_engine.hpp

```diff
@@ -425,6 +425,17 @@ class IntegerNode : public IntegerNodeBase<LeafType> {
 
     size_t find_first_local(size_t start, size_t end) override
     {
+        TConditionFunction c;
+        if (end - start == 1) {
```

Comment:
why not move the `end - start == 1` case into the Array's find_first method itself? Then all call sites will benefit from the optimization.

Reply:
Yes - that might actually be a better idea